src/Plugin/RedirectPlugin.php

```diff
@@ -231,8 +231,34 @@ private function createUri(ResponseInterface $redirectResponse, RequestInterface
         }
 
         $uri = $originalRequest->getUri();
+
+        // Redirections can either use an absolute uri or a relative reference https://www.rfc-editor.org/rfc/rfc3986#section-4.2
+        // If relative, we need to check if we have an absolute path or not
+
+        $path = array_key_exists('path', $parsedLocation) ? $parsedLocation['path'] : '';
+        if (!array_key_exists('host', $parsedLocation) && '/' !== $location[0]) {
```

Comment:
oops. i assumed that `parse_url('')` would fail, but actually it gives an array with path => ''.

i add a check to fail if location is empty, as that would make no sense.